Short titles

simpsonw opened the pull request:
Add the ability to give exhibit pages an alternative title that displays only in the navigation menu.  For example, a page could have a full title of "Late Medieval Catholicism: Lay Piety and Clerical Theology" that's shortened to "Late Medieval Catholicism" in the menu.  This functionality was described in issue #5.

zerocrates commented:
This probably should alter the other nav-related functions/helpers as well: the older "stacked" nav, the previous/next links, the "Up" links... pretty much [this area](https://github.com/omeka/plugin-ExhibitBuilder/blob/master/helpers/ExhibitPageFunctions.php#L70-L261). I could be wrong about the expectation but it does seem appropriate to do this in all those nav-ish places, particularly as the presence of the feature will encourage longer "normal" titles.

There's enough places that use the title for nav-ish stuff that it probably makes sense to implement a dynamic property (using [getProperty](https://github.com/omeka/Omeka/blob/master/application/models/Item.php#L178)) so it can be a simple swap-out of `title` for `nav_title` in `metadata()` calls, where `nav_title` handles the short/regular title fallback you've already got in your single location here.

zerocrates commented:
Beyond that, this looks pretty good to me.

simpsonw commented:
@zerocrates I think I got all the areas you mentioned in ExhibitPageFunctions.php.  I renamed the property to `short_title` (both in the code and the db) since that more clearly describes what it is and added a dynamic property called `menu_title` that returns `short_title` if it's set and not empty and the `title` attribute otherwise.  Please let me know if there's anything else I can do!

zerocrates commented on models/ExhibitPage.php:
```diff
@@ -405,4 +412,24 @@ public function setFixChildrenOnDelete($fix)
+    {
+        switch($property) {
+            case 'menu_title':
+                if(isset($this->short_title) && !empty($this->short_title)) {
```
`!empty` should be sufficient by itself here, as it provides the same error-suppressing properties that `isset` does.

simpsonw replied:
Good point!  I've removed the call to `isset()`.

zerocrates commented:
@mebrett I guess let's just continue to use the PR issue, but I've pulled the feature into master

mebrett commented:
Sounds good, thank you.

mebrett commented:
It works fine except for one issue in Seasons.

If the Exhibit Builder page navigation style for Seasons is set as full-width (under the theme config) then none of the navigation text displays at all. The space is there, but no text.

With short titles: 
![screen shot 2017-07-26 at 4 17 30 pm](https://user-images.githubusercontent.com/2314070/28641704-2fe814be-721e-11e7-84a1-bc01ee8e791b.png)

Older version of Exhibit Builder:
![screen shot 2017-07-26 at 4 19 38 pm](https://user-images.githubusercontent.com/2314070/28641727-3c1cbe9c-721e-11e7-8d6a-ca809826109a.png)

zerocrates commented:
Okay yeah, I missed a couple little errors in the PR: in a few places it tries to access `menu_title` directly as a property, but it needs to be using `metadata()` instead.

zerocrates commented on helpers/ExhibitPageFunctions.php:
```diff
@@ -128,7 +128,7 @@ function exhibit_builder_child_page_nav($exhibitPage = null)
     $html = '<ul class="exhibit-child-nav navigation">' . "\n";
     foreach ($children as $child) {
-        $html .= '<li><a href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $child)) . '">' . html_escape($child->title) . '</a></li>';
+        $html .= '<li><a href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $child)) . '">' . html_escape($child->menu_title) . '</a></li>';
```
The `html_escape($child->menu_title)` here should be `metadata($child, 'menu_title')`

zerocrates commented on helpers/ExhibitPageFunctions.php:
```diff
@@ -92,18 +92,18 @@ function exhibit_builder_page_nav($exhibitPage = null)
             $html .= '<li' . ($pageSibling->id == $page->id ? ' class="current"' : '') . '>';
             $html .= '<a class="exhibit-page-title" href="' . html_escape(exhibit_builder_exhibit_uri($exhibit, $pageSibling)) . '">';
-            $html .= html_escape($pageSibling->title) . "</a></li>\n";
+            $html .= html_escape($pageSibling->menu_title) . "</a></li>\n";
```
should be `metadata($pageSibling, 'menu_title')`

zerocrates commented on helpers/ExhibitPageFunctions.php:
```diff
@@ -92,18 +92,18 @@ function exhibit_builder_page_nav($exhibitPage = null)
     
     foreach ($pagesTrail as $page) {
-        $linkText = $page->title;
+        $linkText = $page->menu_title;
```
This is a problem but it looks like `$linkText` is totally unused anyway so it can just be deleted.

zerocrates commented:
@mebrett OK, I believe that should be fixed

mebrett commented:
@zerocrates Everything looks as it should now, thanks.
